### PR TITLE
CSC Review Fixes - expose delete functions, rename attribute names, add AbstractCache class

### PR DESCRIPTION
**Pull request by dvora-h**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Do tests and lints pass with this change?
- [ ] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [ ] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Is there an example added to the examples folder (if applicable)?
- [ ] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

_Please provide a description of the change here._

### CODE REVIEW

File: redis/_cache.py [hunk 1]
```diff
@@ -160,7 +160,27 @@ class EvictionPolicy(Enum):
     RANDOM = "random"
 
 
-class _LocalCache:
+class AbstractCache:
```
Comment (chayim) [inline redis/_cache.py:163]: Let's docstring to help our users understand why and how to use it.

Comment (chayim) [inline redis/_cache.py:163]: WDYT: Have a look at Abstract base classes, and maybe inherit from [ABC](https://docs.python.org/3/library/abc.html#abc.ABC)?

[the remark this answers is not in the document]
Comment (vladvildanov) [reply redis/_cache.py:163]: https://github.com/vladvildanov/predis/blob/v3.x/src/Cache/CacheInterface.php

[the remark this answers is not in the document]
Comment (vladvildanov) [reply redis/_cache.py:163]: UPD: Forget, I found that OOP works different in Python 😁

**Review by chayim — APPROVED** [review]

**Pull request merged**